test/configs/parse_test.json

```diff
@@ -40,8 +40,57 @@
                         }
 
                     ]
-                }
-            ]
+                },
+                {
+                    "name": "ErrorValueTest",
+                    "id": "ErrorValueTest",
+                    "enabled": true,
+                    "slave_id": "0x91",
+                    "channels": [
+                        {
+                            "name" : "uint16",
+                            "reg_type" : "holding",
+                            "address" : "0x00",
+                            "type": "switch",
+                            "error_value" : 65535
+                        },
+                        {
+                            "name" : "uint32",
+                            "reg_type" : "holding",
+                            "address" : "0x01",
+                            "type": "switch",
+                            "error_value" : 4294967295
+                        },
+                        {
+                            "name" : "str-uint32",
+                            "reg_type" : "holding",
+                            "address" : "0x02",
+                            "type": "switch",
+                            "error_value" : "4294967295"
+                        },
+                        {
+                            "name" : "str-hex-uint32",
+                            "reg_type" : "holding",
+                            "address" : "0x03",
+                            "type": "switch",
+                            "error_value" : "0xFFFFFFFF"
+                        },
+                        {
+                            "name" : "str-uint64",
+                            "reg_type" : "holding",
+                            "address" : "0x04",
+                            "type": "switch",
+                            "error_value" : "18446744073709551615"
+                        },
+                        {
+                            "name" : "str-hex-uint64",
+                            "reg_type" : "holding",
+                            "address" : "0x05",
+                            "type": "switch",
+                            "error_value" : "0xFFFFFFFFFFFFFFFF"
+                        }
+                    ]
+                }            ]
```

Comment:
We need some negative test cases with wrong data (ex. text strings, negative integers etc.)

Reply:
good point. As for negative data, the current implementation uses strtoull() which treats negative values as valid input:

>If the minus sign was part of the input sequence, the numeric value calculated from the sequence of digits is negated as if by unary minus in the result type, which applies unsigned integer wraparound rules.

I actually think it might be useful in rare cases but I'm not sure this behavior deserves to be documented and tested.

P.S. Is it really necessary to speak English here? The only people working on driver are me, you and Islam :)